server/message.go

```diff
@@ -0,0 +1,57 @@
+package server
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/coniks-go/utils"
+)
+
+var (
+	ErrorMalformedClientMessage = errors.New("[coniks] Malformed client request")
+	ErrorNameExisted            = errors.New("[coniks] Registering identity is already registered")
+	ErrorInternalServer         = errors.New("[coniks] Internal server error")
+)
+
+const (
+	RegistrationType = "registration"
+)
+
+type Request struct {
+	Type    string
+	Request interface{}
+}
+
+type RegistrationRequest struct {
+	Username               string `json:"username"`
+	Key                    string `json:"key"`
+	KeyType                string `json:"key_type,omitempty"`
+	AllowUnsignedKeychange bool   `json:"allow_unsigned_key_change,omitempty"`
+	AllowPublicLookup      bool   `json:"allow_public_lookup,omitempty"`
+}
+
+type Response interface{}
+
+type ErrorResponse struct {
+	Error string
+}
+
+type RegistrationResponse struct {
+	TempraryBinding string `json:"tb"`
+}
+
+func NewErrorResponse(err error) *ErrorResponse {
+	return &ErrorResponse{Error: err.Error()}
+}
+
+func (reg *RegistrationRequest) serialize() []byte {
+	// serialize scheme:
```

Comment:
> Maybe something like messaging.go could work?

But we already have `message.go`. I think we can keep that name and move `serialize()` to something like `messagekv.go`, like what we are doing in `db` branch. What do you think?